src/lib/testrail.ts

```diff
@@ -83,20 +94,66 @@ export class TestRail {
       },
       data: JSON.stringify({ results }),
     })
-      .then(response => {
-        console.log('\n', chalk.magenta.underline.bold('(TestRail Reporter)'));
-        console.log(
-          '\n',
-          ` - Results are published to ${chalk.magenta(
-            `${this.options.host}/index.php?/runs/view/${this.runId}`
-          )}`,
-          '\n'
-        );
-      })
-      .catch(error => console.error(error));
+    .then(response => {
+      return response.data
+    })
+    .catch(error => { 
+      return console.error(error); 
+    });
   }
 
+  // This is an API call to the TestRail with proper attachment
+  public loadAttachment (resultId, attachment) {
+    const options = {
+      method: "POST",
+      url: this.base + "/add_attachment_to_result/" + resultId,
+      headers: {
+          "Content-Type": "multipart/form-data"
+      },
+      auth: {
+          username: this.options.username,
+          password: this.options.password,
+      },
+      formData: {
+          "attachment": fs.createReadStream(`./${attachment}`)
+      }
+    };
+
+    request(options, function (err) {
+      if (err) console.log(err);
+    })
+  }
+
+  // This function will attach failed screenshot on each test result(comment) if founds it
+  public addAttachmentToResult (results, loadedResultId) {
+  var caseId = results[0].case_id
+    try {
+      find.file('./cypress/screenshots/', (files) => {
+        files.filter(file => file.includes(`C${caseId}`)).forEach(screenshot => {
+          if(this.counter === 1) { 
+            if (screenshot.includes('(failed).png') === true) {
+              this.loadAttachment(loadedResultId, screenshot)
+              this.counter++
+            }
+          } else if (this.counter === 2) {
+            if (screenshot.includes('(attempt 2).png') === true) {
+              this.loadAttachment(loadedResultId, screenshot)
+              this.counter++
+            }
+          } else {
+            if (screenshot.includes('(attempt 3).png') === true) {
+              this.loadAttachment(loadedResultId, screenshot)
+            }
+          }
```

Comment:
This whole approach with setting `this.counter` and nested `if`s is a bit hacky. Instead of setting counter you can just use `String.prototype.match()` and test it against a regex to see if there is `failed` or `attempt` in the screenshot name.

Here is the reference how `match()` method works: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/match

And this is great site to use to test if regex will match your string and learn more about regex in general: https://regexr.com/